src/instructlab/eval/mt_bench_branch_generator.py

```diff
@@ -23,6 +23,9 @@ def get_file_paths(directory):
     logger.debug(locals())
     file_paths = []
     for root, _, files in os.walk(directory):
+        if root.startswith(os.path.join(directory, "knowledge")):
```

Comment:
I'm always confused about how `os.walk` is supposed to work cross-platform (e.g on Windows). Are we sure that `os.path.join(directory, N)` always equals to `root`? This may ring a bell: https://stackoverflow.com/questions/61252884/why-does-root-returned-from-os-walk-contain-as-directory-separator-but-os-se

I don't have access to win32 to check how this would work. I also am not sure if we care. But leaving it here just in case.

Reply:
It looks like reworking this to use pathlib.Path might fix this? At least according to that thread...

Reply:
Yes, `Path()` should probably normalize everything for you. Note that I don't think python3.11 has the native `Path.walk` so you'll have to deal with both. (first `os.walk`, then `Path(root)`...)

Reply:
btw consider that the `directory`, despite some typing claims otherwise, comes from `ilab eval` where the CLI option is defined as `type=click.Path()`. I think it means that `directory` (may) arrive here as `Path`. Not sure if it affects your alternative version (if you decide to have one.)